### PR TITLE
Document JS test suite basics

### LINKED ISSUE #15951: Document JS test suite basics
Similar to the test-specific README in https://github.com/twbs/bootlint/tree/master/test , I propose we add some docs explaining the basics of the test suite. E.g. we use QUnit (*link to QUnit's API docs*), there's one file per JS plugin, here's how to run the unit tests, here's what the "visual" tests are, etc.

### PR DESCRIPTION
**Pull request by cvrebert**
This is just a tweaked version of #15952.
Fixes #15951.
CC: @hnrch02 for final review

### CODE REVIEW

File: js/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      + '<a class="close" href="#" data-dismiss="alert">×</a>'
+      + '<p><strong>Template necessary for the test.</p>'
+      + '</div>'
+  var $alert = $(templateHTML).bootstrapAlert()
+
+  $alert.find('.close').click()
+
+  // Make assertion
+  assert.strictEqual($alert.hasClass('in'), false, 'remove .in class on .close click')
+})
```
Comment (hnrch02) [inline js/tests/README.md]: Maybe a async example could also be of value?

File: js/tests/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+## How does Bootstrap's test suite work?
+
+Bootstrap uses [QUnit](http://api.qunitjs.com/), a powerful, easy-to-use JavaScript unit test framework. Each plugin has a file dedicated to its tests in `unit/<plugin-name>.js`.
+
+* `unit/` contains the unit test files for each Bootstrap plugin.
+* `vendor/` contains third-party testing-related code for the browser environment (jQuery and QUnit).
```
Comment (hnrch02) [inline js/tests/README.md]: These files are also used when testing in PhantomJS so the remark about the browser environment isn't accurate.

File: js/tests/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+To run the unit test suite via a real web browser, open `index.html` in the browser.
+
+
+## How do I add a new unit test?
+
+1. Locate and open the file dedicated to the plugin which you need to add tests to (`unit/<plugin-name>.js`).
+2. Review the [QUnit API Documentation](http://api.qunitjs.com/) and use the existing tests as references for how to structure your new tests.
+3. Write the necessary unit test(s) for the new or revised functionality.
+4. Run `grunt test-js` to see the results of your newly-added test(s). They should fail before your changes are applied to the plugin, and should pass after your changes are applied to the plugin.
```
Comment (hnrch02) [inline js/tests/README.md]: Maybe we could emphasize more that new tests need to fail without the "new or revised functionality".

File: js/tests/README.md (hunk 2 above)
Comment (cvrebert) [inline js/tests/README.md]: Fixed.

File: js/tests/README.md (hunk 1 above)
Comment (cvrebert) [inline js/tests/README.md]: Agreed.

**Comment by kkirsche** [comment]
Thanks @cvrebert :smiley: I'm sure there were plenty of small little things that needed tweaking. Thank you for your time and assistance though on getting this in :+1:

File: js/tests/README.md (hunk 1 above)
Comment (cvrebert) [inline js/tests/README.md]: Added async example

**Comment by cvrebert** [comment]
@kkirsche Thanks again

**Comment by kkirsche** [comment]
My pleasure. I'm glad I was able to help with it :smiley: I appreciate everyone's patience as it was fixed and finalized.

**Pull request merged**